packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -6555,6 +6556,302 @@ void main() {
+
+      final EditableTextState state = tester.firstState(find.byType(EditableText));
+      const String jsonInsertionDelta = '{'
+          '"oldText": "",'
+          ' "deltaText": "let there be text",'
+          ' "deltaType" : "TextEditingDeltaType.insertion",'
+          ' "deltaStart": 0,'
+          ' "deltaEnd": 0,'
+          ' "selectionBase": 17,'
+          ' "selectionExtent": 17,'
+          ' "selectionAffinity" : "TextAffinity.downstream" ,'
+          ' "selectionIsDirectional": false,'
+          ' "composingBase": -1,'
+          ' "composingExtent": -1}';
+
+      const String jsonDeletionDelta = '{'
+          '"oldText": "let there be text",'
+          ' "deltaText": " text",'
+          ' "deltaType" : "TextEditingDeltaType.deletion",'
+          ' "deltaStart": 12,'
+          ' "deltaEnd": 17,'
+          ' "selectionBase": 12,'
+          ' "selectionExtent": 12,'
+          ' "selectionAffinity" : "TextAffinity.downstream" ,'
+          ' "selectionIsDirectional": false,'
+          ' "composingBase": -1,'
+          ' "composingExtent": -1}';
+
+      const String jsonReplacementDelta = '{'
+          '"oldText": "let there be",'
+          ' "deltaText": "be light",'
+          ' "deltaType" : "TextEditingDeltaType.replacement",'
+          ' "deltaStart": 10,'
+          ' "deltaEnd": 12,'
+          ' "selectionBase": 18,'
+          ' "selectionExtent": 18,'
+          ' "selectionAffinity" : "TextAffinity.downstream" ,'
+          ' "selectionIsDirectional": false,'
+          ' "composingBase": -1,'
+          ' "composingExtent": -1}';
+
+      const String jsonEqualityDelta = '{'
+          '"oldText": "let there be light",'
+          ' "deltaText": "",'
+          ' "deltaType" : "TextEditingDeltaType.equality",'
+          ' "deltaStart": -1,'
+          ' "deltaEnd": -1,'
+          ' "selectionBase": 18,'
+          ' "selectionExtent": 18,'
+          ' "selectionAffinity" : "TextAffinity.downstream",'
+          ' "selectionIsDirectional": false,'
+          ' "composingBase": -1,'
+          ' "composingExtent": -1}';
+
+      final TextEditingDelta insertionDelta = TextEditingDelta.fromJSON(jsonDecode(jsonInsertionDelta) as Map<String, dynamic>);
+      final TextEditingDelta deletionDelta = TextEditingDelta.fromJSON(jsonDecode(jsonDeletionDelta) as Map<String, dynamic>);
+      final TextEditingDelta replacementDelta = TextEditingDelta.fromJSON(jsonDecode(jsonReplacementDelta) as Map<String, dynamic>);
+      final TextEditingDelta equalityDelta = TextEditingDelta.fromJSON(jsonDecode(jsonEqualityDelta) as Map<String, dynamic>);
+
+      state.updateEditingValueWithDeltas(<TextEditingDelta>[insertionDelta, deletionDelta, replacementDelta, equalityDelta]);
+      await tester.pump();
+      expect(controller.text, 'let there be light');
+      expect(controller.selection, equalityDelta.selection);
+      expect(state.currentTextEditingValue.composing, equalityDelta.composing);
+    });
+  });
+
```

Comment:
Do we have any integration tests that test text editing?  If so, maybe as an easy additional test for this, we could run them a second time with `enableDeltaModel` set to true, and make sure that text editing still works as normal.  Not sure if we have any though.